packages/cw-derive/examples/generics.rs

```diff
@@ -0,0 +1,16 @@
+#![allow(dead_code)]
+use cosmwasm_std::{CosmosMsg, Response};
+
+pub struct Ctx;
+pub struct Error;
+
+#[cw_derive::interface(module=msg, exec=Cw1Exec, query=Cw1Query)]
```

Comment:
It is hard to check what these macros actually expand into.

Maybe a simple test case that constructs a valid ExecuteMsg.

```rust
#[cfg(test)]
mod test {
    #[derive(Debug, PartialEq, Clone, JsonSchema)]
    struct MyMsg {
        pub foo: String;
    }

    #[test]
    // this just ensures that the autogenerated message types can be constructed as expected
    fn check_msg() {
        let exec = Cw1Exec<MyMsg>::Execute{
           msgs: vec![
               BankMsg::Burn{ amount: coins(500, "atom") }.into(), 
               CosmosMsg::Custom(MyMsg{ foo: "hello world".to_string() }),
           ],
        };
        cosmwasm_std::to_binary(&exec).unwrap();
    }
}
```

Reply:
Next thing I am working right now are actual dispatch functions, and there would be tests like you shown, including checking if everythign works on this stage.

I always check what is generated with `cargo expand` - being honest I work with it all the time fixing small generation issues, but I agree it is not good enough that I know what is generated, tests are needed. I just wanted to have API allowing to test behavior on any level (so next small step will include real tests).